cypress-io/cypress
# fix: allow asynchronous vue cli init phase

- Closes [issue-23896](https://github.com/cypress-io/cypress/issues/23896)

### User facing changelog

n/a

### Additional details
The function `vueCliHandler` in `webpack-dev-server/src/helpers/vueCLIHandlers.ts` is now asynchronous and re-implements  [vue-cli -> webpack.config.js](https://github.com/vuejs/vue-cli/blob/dev/packages/%40vue/cli-service/webpack.config.js) with a fix to wait for the promise returned by `init` if the init-phase was asynchronous.  


### Steps to test
Please see information in [issue-23896](https://github.com/cypress-io/cypress/issues/23896) for a test setup. 
In the test setup run `./node_modules/.bin/cypress run --component` to execute the component test.

### How has the user experience changed?
Now adding a `vue.config.js` file to a repository with `"type": "module"` in `package.json` does not crash because the compiler is missing par.

### PR Tasks

- [ ] Have tests been added? (I was unable to grasp how I would add a test which uses the configuration given in [issue-23896](https://github.com/cypress-io/cypress/issues/23896))
- [x] Have tests been updated?
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)?
